src/mscorlib/shared/System/ReadOnlyMemory.cs

```diff
@@ -67,22 +70,21 @@ public ReadOnlyMemory(T[] array, int start, int length)
             if ((uint)start > (uint)array.Length || (uint)length > (uint)(array.Length - start))
                 ThrowHelper.ThrowArgumentOutOfRangeException();
 
-            _arrayOrOwnedMemory = array;
+            _object = array;
             _index = start;
             _length = length;
         }
-        
-        // Constructor for internal use only.
+
+        /// <summary>Creates a new memory over the existing object, start, and length.  No validation is performed.</summary>
+        /// <param name="obj">The target object.</param>
+        /// <param name="start">The index at which to begin the memory.</param>
+        /// <param name="length">The number of items in the memory.</param>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        internal ReadOnlyMemory(OwnedMemory<T> owner, int index, int length)
+        internal ReadOnlyMemory(object obj, int start, int length)
         {
-            if (owner == null)
-                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.ownedMemory);
-            if (index < 0 || length < 0)
-                ThrowHelper.ThrowArgumentOutOfRangeException();
-
-            _arrayOrOwnedMemory = owner;
-            _index = index | (1 << 31); // Before using _index, check if _index < 0, then 'and' it with RemoveOwnedFlagBitMask
+            // No validation performed
+            _object = obj;
+            _index = start;
```

Comment:
Without `_index = index | (1 << 31);` for OwnedMemory, how would we be able to distinguish between T[] and OwnedMemory?

Reply:
OwnedMemory is never used to create a ReadOnlyMemory, only Memory.  And Memory creates a ReadOnlyMemory via a reinterpret cast, so that code wasn't used.  Have I misunderstood?  If it's a problem, we have a test whole, as all tests are passing.

Reply:
Ah yes, good point.

Although, I recall we had less than 100% branch coverage for tests, specifically for the internal constructor:
https://user-images.githubusercontent.com/6527137/29992954-49b3bce8-8f5d-11e7-8871-f26a77451a7f.png